pocean/dsg/profile/im.py

```diff
@@ -166,7 +167,8 @@ def calculated_metadata(self, df=None, geometries=True, clean_cols=True, clean_r
         first_row = df.iloc[0]
         first_loc = Point(first_row.x, first_row.y)
         if geometries:
-            coords = list(unique_justseen(zip(df.x, df.y)))
+            coords = unique_justseen(zip(df.x.tolist(), df.y.tolist()))
```

Comment:
@ocefpaf  I'm going to go ahead and contradict myself. Just tested on a huge glider deployment.

How's this? Passes the tests.

```python
# only extract non-null pairs
null_coordinates = tgroup.x.isnull() | tgroup.y.isnull()
coords = list(unique_justseen(zip(tgroup.x[~null_coordinates].tolist(), tgroup.y[~null_coordinates].tolist())))
```